synapse/config/oidc.py

```diff
@@ -280,6 +280,20 @@ def _parse_oidc_config_dict(
         for x in oidc_config.get("attribute_requirements", [])
     ]
 
+    # Read from either `client_secret_path` or `client_secret`. If both exist, error.
+    client_secret = oidc_config.get("client_secret")
+    client_secret_path = oidc_config.get("client_secret_path")
+    if client_secret_path is not None:
+        if client_secret is None:
+            client_secret = read_file(
+                client_secret_path, config_path + ("client_secret_path",)
+            ).rstrip("\n")
```

Comment:
I think this can be simplified with `self.read_file`?

```suggestion
            client_secret = self.read_file(client_secret_file)
```

Reply:
@clokep  AFAICS this isn't in a class context and there's no `self` defined. Also, the `read_file` implementation I found doesn't strip the final `\n` so in case this is part of the file, it will be added to the `client_secret` string. Been bitten by that in other software already :)

Reply:
Oh right, this is a free function, sorry about that!